OpenStack-Rabbit-Consumer/rabbit_consumer/message_consumer.py

```diff
@@ -1,281 +1,134 @@
 import json
 import logging
-import socket
-from typing import Optional
+from typing import Optional, List
 
 import rabbitpy
 
 from rabbit_consumer import aq_api
 from rabbit_consumer import openstack_api
 from rabbit_consumer.aq_api import verify_kerberos_ticket
 from rabbit_consumer.consumer_config import ConsumerConfig
-from rabbit_consumer.aq_fields import AqFields
-from rabbit_consumer.os_descriptions import OsDescription
+from rabbit_consumer.openstack_address import OpenstackAddress
+from rabbit_consumer.os_descriptions.os_descriptions import OsDescription
+from rabbit_consumer.rabbit_message import RabbitMessage
+from rabbit_consumer.vm_data import VmData
 
 logger = logging.getLogger(__name__)
 
 
-def is_aq_managed_image(message) -> Optional[OsDescription]:
+def is_aq_managed_image(rabbit_message: RabbitMessage) -> Optional[OsDescription]:
     """
     Check to see if the metadata in the message contains entries that suggest it
     is for an Aquilon VM.
     """
-    payload = message.get("payload", None)
-    if not payload:
-        logger.debug("No payload found in message")
-        return None
 
-    image_name = payload.get("image_name", None)
+    image_name = rabbit_message.payload.image_name
     if not image_name:
         logger.debug("No image_name found in payload")
         return None
 
-    assert isinstance(image_name, str)
     try:
         return OsDescription.from_image_name(image_name)
     except ValueError:
         logger.debug("Image name %s does not match an Aquilon image name", image_name)
         return None
 
 
-def get_metadata_value(message, key):
-    """
-    Function which gets the value from the possible for a given metadata key
-    from the possible paths in the image or instance metadata with
-    the key in uppercase or lowercase
-    """
-    returnstring = message.get("payload").get("metadata").get(key)
-    if returnstring is None:
-        returnstring = message.get("payload").get("image_meta").get(key)
-        if returnstring is None:
-            returnstring = message.get("payload").get("metadata").get(key.lower())
-            if returnstring is None:
-                returnstring = message.get("payload").get("image_meta").get(key.lower())
-    return returnstring
-
+def consume(message: RabbitMessage):
```

Comment:
Could we add a description for this method and the type of variable is returned?